Indicators/PivotPointsHighLow.cs

```diff
@@ -0,0 +1,247 @@
+
+        /// <summary>
+        /// Creates a new instance of <see cref="PivotPointsHighLow"/> indicator with an equal high and low length
+        /// </summary>
+        /// <param name="length">The length parameter here defines the number of surround bars that we compare against the current bar high and lows for the max/min </param>
+        public PivotPointsHighLow(int length)
+            : this($"PivotPointsHighLow({length})", length, length)
+        { }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="PivotPointsHighLow"/> indicator
+        /// </summary>
+        /// <param name="lengthHigh">The number of surrounding bars whose high values should be less than the current one's for the high pivot point be assigned</param>
+        /// <param name="lengthLow">The number of surrounding bars whose lows values should be less than the current one's for the low pivot point be assigned</param>
+        public PivotPointsHighLow(int lengthHigh, int lengthLow)
+            : this($"PivotPointsHighLow({lengthHigh},{lengthLow})", lengthHigh, lengthLow)
+        { }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="PivotPointsHighLow"/> indicator
+        /// </summary>
+        /// <param name="name">The name of an indicator</param>
+        /// <param name="lengthHigh">The number of surrounding bars whose high values should be less than the current one's for the high pivot point be assigned</param>
+        /// <param name="lengthLow">The number of surrounding bars whose lows values should be less than the current one's for the low pivot point be assigned</param>
+        public PivotPointsHighLow(string name, int lengthHigh, int lengthLow) :
+            base(name, 2 * Math.Max(lengthLow, lengthHigh) + 1)
+        {
+            _lengthHigh = lengthHigh;
+            _lengthLow = lengthLow;
+        }
+
+        /// <inheritdoc />
+        protected override decimal ComputeNextValue(IReadOnlyWindow<IBaseDataBar> window, IBaseDataBar input)
+        {
+            if (!IsReady) return 0m;
+
+            var middlePoint1 = window[_lengthHigh];
+            var isHigh = false;
+            for (var k = 0; k < (2 * _lengthHigh + 1); k++)
+            {
+                // Skip the middle point
+                if (k == _lengthHigh) continue;
+
+                // Check if current high is below middle point high
+                isHigh = window[k].High < middlePoint1.High;
+
+                // If any high is not below middle point high -> this is not a pivot point
+                if (!isHigh) break;
+            }
+
+            var middlePoint2 = window[_lengthLow];
+            var isLow = false;
+            for (var k = 0; k < (2 * _lengthLow + 1); k++)
+            {
+                if (k == _lengthLow) continue;
+
+                isLow = window[k].Low > middlePoint2.Low;
+
+                if (!isLow) break;
+            }
```

Comment:
Think it would be nice to unit test this logic, could create a test `PivotPointsHighLow` which exposes this method to call it directly and assert the behavior. 
I find this `var middlePoint2 = window[_lengthLow];` a bit suspicious it's indirectly relating the high & low length through the window period. For example, what if `_lengthHigh` is 1000 & `_lengthLow` is 1,  with `window[_lengthLow]` we would be reading old values in the window

Reply:
This has been fixed in 6e13947 -> Implements separate rolling windows to calculate highs and low. You are correct the logic should be separate for highs and lows, as they may vary significantly in length